### PR TITLE
Upgrading to latest code-generator commit

### PR DESCRIPTION
**Pull request by AaronME**
Signed-off-by: Aaron Eaton <aaron@upbound.io>

### Description of your changes

Bumping provider-aws to latest code-generator commit.

Fixes https://github.com/crossplane/provider-aws/issues/876

I have:

- [X] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

This PR is currently not mergeable. https://github.com/aws-controllers-k8s/code-generator/pull/237 has broken some of our functionality, including:
- The APIGroupSuffix is no longer being set to 'aws.crossplane.io`
- The Package Metadata in each service is now being ignored. This is resulting in everything being re-built for the latest apiVersion, instead of respecting the multiple apiVersions we support.

Will find some time to dig into it and submit another fix upstream.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/apigatewayv2/v1beta1/zz_api.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+// +kubebuilder:object:root=true
+
+// API is the Schema for the APIS API
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="EXTERNAL-NAME",type="string",JSONPath=".metadata.annotations.crossplane\\.io/external-name"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,aws}
+type API struct {
```
Comment (muvaf) [inline apis/apigatewayv2/v1beta1/zz_api.go:100]: Do you know why these are created in `v1beta1` version?

**Comment by EdgeJ** [comment]
@AaronME after pushing that PR yesterday, I realized that the function that checks for the latest api doesn't work correctly on crossplane and that there is a variable that sets the api version instead.  I added a Pr to fix that and use the supplied variable here: https://github.com/aws-controllers-k8s/code-generator/pull/240  I think this would at least fix some, if not all, of the issues seen here--particularly resources being created with `v1beta1` apis.

**Comment by Dkaykay** [comment]
Any updates on the status?

File: apis/apigatewayv2/v1beta1/zz_api.go (hunk 1 above)
Comment (AaronME) [reply apis/apigatewayv2/v1beta1/zz_api.go]: @muvaf This was fixed by https://github.com/aws-controllers-k8s/code-generator/pull/240

**Comment by AaronME** [comment]
Closing and re-opening to trigger tests.

**Comment by AaronME** [comment]
This PR is currently blocked by the issue described in https://github.com/aws-controllers-k8s/community/issues/1078 and https://github.com/aws-controllers-k8s/community/issues/1065

We are tracking those issues for the fix and will update as soon as they are available.

**Comment by muvaf** [comment]
@AaronME I didn't notice this PR before opening https://github.com/crossplane/provider-aws/pull/993 . Do you want to continue with this one after a rebase?

File: apis/generate.go [hunk 2]
```diff
@@ -28,9 +28,14 @@ limitations under the License.
 // Generate crossplane-runtime methodsets (resource.Claim, etc)
 //go:generate go run -tags generate github.com/crossplane/crossplane-tools/cmd/angryjet generate-methodsets --header-file=../hack/boilerplate.go.txt ./...
 
+// TODO(muvaf): Remove the last mockgen usage in Addon resource and move to
+// current convention with tests.
+//go:generate go run github.com/golang/mock/mockgen  --build_flags=--mod=mod -package eksiface -copyright_file ../hack/boilerplate.go.txt -destination ../pkg/clients/eks/fake/eksiface/fake.go github.com/aws/aws-sdk-go/service/eks/eksiface EKSAPI
```
Comment (muvaf) [inline apis/generate.go:33]:
```suggestion
//go:generate go run github.com/golang/mock/mockgen  --build_flags=--mod=mod -package eksiface -copyright_file ../hack/boilerplate.go.txt -destination ../pkg/clients/eks/fake/eksiface/zz_fake.go github.com/aws/aws-sdk-go/service/eks/eksiface EKSAPI
```

**Review by muvaf — APPROVED** [review]
LGTM! Thanks a lot @AaronME for tracking the fixes that made this upgrade possible! I'm labeling the PR as `breaking-change` so that we don't forget to add instructions to the release notes.

**Pull request merged**